lib/govspeak/remove_advisory_service.rb

```diff
@@ -0,0 +1,101 @@
+module Govspeak
+  class RemoveAdvisoryService
+    attr_reader :body
+
+    def initialize(object)
+      @object = object
+      @body = object.body || object.govspeak_content.body
+      @whodunnit = User.find_by(name: "GDS Inside Government Team")
+    end
+
+    def process!
+      if @object.is_a?(Edition)
+        AuditTrail.acting_as(@whodunnit) do
+
+          # Create a new draft of the edition
+          draft = @object.create_draft(@whodunnit)
+
+          # Replace advisories in the body of the new attachment
+          new_body = replace_all_advisories(body)
+
+          # Update the draft edition and publish
+          draft.update!(
+            body: new_body,
+            minor_change: true,
+          )
+          draft.submit!
+          publish_reason = "Replacing deprecated advisory elements with information callouts"
+          edition_publisher = Whitehall.edition_services.publisher(draft, user: @whodunnit, remark: publish_reason)
+          edition_publisher.perform!
+        end
+      elsif @object.is_a?(HtmlAttachment)
+        AuditTrail.acting_as(@whodunnit) do
+          # Create a draft of the attachable edition
+          draft = @object.attachable.create_draft(@whodunnit)
+
+          # Find the attachment in the new draft
+          new_attachment = draft.html_attachments.find_by(slug: @object.slug)
+
+          # Replace advisories in the body of the new attachment
+          new_body = replace_all_advisories(new_attachment.body)
+          new_attachment.govspeak_content.update!(body: new_body)
+
+          # Update the draft edition and publish
+          draft.update!(minor_change: true)
+          draft.submit!
```

Comment:
You could probably DRY up the `process!` method by creating a ` submit_and_publish(draft)`